HZETRN/btrn.py

```diff
@@ -0,0 +1,377 @@
+	###################################
+	# For backscatter:  
+	# Create two lists 'a' and 'b'.  
+	#    'a' is the list of slab materials and lengths in the given direction.  
+	#    'b' is the list of same items going OPPOSITE (180 degrees)
+	print start_vol, ref_point, "dir", dir
+	slab_lens_a, slab_mat_names_a = slabs_for_ray(start_vol, ref_point,  dir, vol_fname_dict)
+	slab_lens_b, slab_mat_names_b = slabs_for_ray(start_vol, ref_point, -dir, vol_fname_dict)
+
+	# reverse order and add backward dir
+        # slab_lens = slab_lens_a[::-1] + slab_lens_b
+        slab_lens      = slab_lens_b[::-1]      + slab_lens_a
+	slab_mat_names = slab_mat_names_b[::-1] + slab_mat_names_a
+	num_slabs_to_ref = len(slab_lens_b)
+	num_mats = len(slab_mat_names)
+        #############################################
+	# Create the transport geometry file contents
+        # Need to have a non-zero number of mats
+	transport_input = []
+	if 0 != num_mats:
+	    effective_num_mats = num_mats
+	    last_mat_name = slab_mat_names[0]
+	    last_slab_points = [0.0, slab_lens[0]]
+
+	    for n in range(1,num_mats):
+                if slab_mat_names[n] == last_mat_name:
+		    #  Need to ADD the two lengths together: they are cumulative
+		    last_slab_points.append(slab_lens[n] + slab_lens[n-1])
+		    effective_num_mats = effective_num_mats - 1
+		else:
+	            transport_input.append(last_mat_name)
+	            transport_input.append(str(len(last_slab_points)))
+		    slab_points_list = ["{0:.1f}".format(x) for x in last_slab_points]
+		    transport_input.append(", ".join(slab_points_list))
+                    # Now re-initialize to the current mat/length
+		    last_mat_name = slab_mat_names[n]
+		    last_slab_points = [0.0, slab_lens[n]]
+	    # The last slab got set up but not written
+	    transport_input.append(last_mat_name)
+	    transport_input.append(str(len(last_slab_points)))
+	    slab_points_list = ["{0:.1f}".format(x) for x in last_slab_points]
+	    transport_input.append(", ".join(slab_points_list))
+	        
+	    transport_input.insert(0,(str(effective_num_mats)))
+
+	# No materials for this or the opposite direction: make bogus spatial file 
+	else:
+	    transport_input.append('0')
+	    transport_input.append('No material traversed, either direction')
+	    transport_input.append('2')
+	    transport_input.append('0.0 0.0')
+
+	# Need a newline at end of file
+	transport_input.append('\n')
+
+	dir_string = "{0:.4f}".format(dir[0]) + '_' + \
+	             "{0:.4f}".format(dir[1]) + '_' + \
+		     "{0:.4f}".format(dir[2]) 
+
+	# ToDo: change the base name here and on the other side
+	if len(ray_tuples) < 20:
+	    spatial_filename = 'spatial_ba' + dir_string + '.dat'
+	    # sslab = []
+            # for d in slab_lengths:
+	    #     sslab.append(format(d,'.6f'))
+	    # print ('dist, mats ', sslab, slab_mat_names)
+            #############################################
+	else:
+	    spatial_filename = 'spatial.dat'
```

Comment:
Actually not.  The directory containing 'spatial.dat' or spatial_ba_....dat is the global.